### DISCUSSION TITLE
`yarn rebuild:browser` success, but `yarn rebuild:electron` fail with 'Could not find any Visual Studio installation to use'

### DISCUSSION THREAD

**Discussion by davidxwwang** [eclipse-emfcloud/emfcloud#202] [Coffee Editor] [answered]
hi, l download the code ,and `cd client`, then use 'yarn 'and 'yarn start', and l can see the web ide on `127.0.0.1:3000`, later l want a destop version , so l add a workspace named 'electron-app', and the necessary configurations，  the package.json'in the electron-app workspace like this:
`{
  "private": true,
  "name": "electron-app",
  "main": "src-gen/frontend/electron-main.js",
  "license": "(EPL-2.0 OR MIT)",
  "version": "0.8.0",
  "contributors": [
    {
      "name": "Eclipse EMF.cloud Project",
      "email": "emfcloud-dev@eclipse.org",
      "url": "https://projects.eclipse.org/projects/ecd.emfcloud"
    }
  ],
  "repository": {
    "type": "git",
    "url": "https://github.com/eclipse-emfcloud/coffee-editor.git"
  },
  "homepage": "https://www.eclipse.org/emfcloud/",
  "bugs": "https://github.com/eclipse-emfcloud/coffee-editor/issues",
  "dependencies": {
    "@theia/electron": "latest",
    "@theia/callhierarchy": "^1.34.1",
    "@theia/console": "^1.34.1",
    "@theia/core": "^1.34.1",
    "@theia/debug": "^1.34.1",
    "@theia/editor": "^1.34.1",
    "@theia/file-search": "^1.34.1",
    "@theia/filesystem": "^1.34.1",
    "@theia/git": "^1.34.1",
    "@theia/keymaps": "^1.34.1",
    "@theia/markers": "^1.34.1",
    "@theia/messages": "^1.34.1",
    "@theia/mini-browser": "^1.34.1",
    "@theia/monaco": "^1.34.1",
    "@theia/navigator": "^1.34.1",
    "@theia/outline-view": "^1.34.1",
    "@theia/output": "^1.34.1",
    "@theia/plugin-dev": "^1.34.1",
    "@theia/plugin-ext": "^1.34.1",
    "@theia/plugin-ext-vscode": "^1.34.1",
    "@theia/plugin-metrics": "^1.34.1",
    "@theia/preferences": "^1.34.1",
    "@theia/process": "^1.34.1",
    "@theia/scm": "^1.34.1",
    "@theia/scm-extra": "^1.34.1",
    "@theia/search-in-workspace": "^1.34.1",
    "@theia/task": "^1.34.1",
    "@theia/terminal": "^1.34.1",
    "@theia/typehierarchy": "^1.34.1",
    "@theia/userstorage": "^1.34.1",
    "@theia/variable-resolver": "^1.34.1",
    "@theia/vsx-registry": "^1.34.1",
    "@theia/workspace": "^1.34.1",
    "coffee-common": "^0.8.0",
    "coffee-comparison": "^0.8.0",
    "coffee-cpp-extension": "^0.8.0",
    "coffee-editor-extension": "^0.8.0",
    "coffee-java-extension": "^0.8.0",
    "coffee-servers": "^0.8.0",
    "coffee-welcome-page": "^0.8.0",
    "coffee-workflow-analyzer": "^0.8.0",
    "coffee-workflow-glsp-theia": "^0.8.0"
  },
  "devDependencies": {
    "@theia/cli": "^1.34.1",
    "electron-builder": "^22.4.1",
    "electron": "^15.3.5"
  },
  "scripts": {
    "prepare": "",
    "development": "theia build --mode development",
    "production": "theia build --mode production",
    "start": "theia start",
    "watch": "theia build --watch --mode development"
  },
  "theia": {
    "target": "electron",
    "frontend": {
      "config": {
        "applicationName": "emf.cloud Coffee Editor",
        "defaultIconTheme": "theia-file-icons",
        "preferences": {
          "security.workspace.trust.startupPrompt": "never",
          "security.workspace.trust.enabled": false,
          "files.autoSave": "off"
        }
      }
    }
  }
}
`
then 
(1) using `yarnrebuild:browser`, tip: 'native node modules are already rebuilt for browser'
(2) using `yarn rebuild:electron` tip errors:'Could not find any Visual Studio installation to use' following:
`
$ theia rebuild:browser
native node modules are already rebuilt for browser
Done in 0.79s.
PS C:\Users\surface\Desktop\che-component-with-git\coffee-editor-master-0314\client> yarn rebuild:electr
on
yarn run v1.22.19
$ theia rebuild:electron
Processed "native-keymap"
Processed "find-git-repositories"
Processed "drivelist"
Processed "node-pty"
Processed "nsfw"
✖ Rebuild Failed
An unhandled error occurred inside electron-rebuild
node-gyp failed to rebuild 'C:\Users\surface\Desktop\che-component-with-git\coffee-editor-master-0314\cl
ient\node_modules\drivelist'.
For more information, rerun with the DEBUG environment variable set to "electron-rebuild".

Error: Could not find any Visual Studio installation to use

Error: node-gyp failed to rebuild 'C:\Users\surface\Desktop\che-component-with-git\coffee-editor-master-
0314\client\node_modules\drivelist'.
For more information, rerun with the DEBUG environment variable set to "electron-rebuild".

Error: Could not find any Visual Studio installation to use

    at NodeGyp.rebuildModule (C:\Users\surface\Desktop\che-component-with-git\coffee-editor-master-0314\
client\node_modules\electron-rebuild\lib\src\module-type\node-gyp.js:120:19)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
    at async ModuleRebuilder.rebuildNodeGypModule (C:\Users\surface\Desktop\che-component-with-git\coffe
e-editor-master-0314\client\node_modules\electron-rebuild\lib\src\module-rebuilder.js:98:9)
    at async ModuleRebuilder.rebuild (C:\Users\surface\Desktop\che-component-with-git\coffee-editor-mast
er-0314\client\node_modules\electron-rebuild\lib\src\module-rebuilder.js:128:14)
    at async Rebuilder.rebuildModuleAt (C:\Users\surface\Desktop\che-component-with-git\coffee-editor-ma
ster-0314\client\node_modules\electron-rebuild\lib\src\rebuild.js:149:13)
    at async Rebuilder.rebuild (C:\Users\surface\Desktop\che-component-with-git\coffee-editor-master-031
4\client\node_modules\electron-rebuild\lib\src\rebuild.js:112:17)
    at async C:\Users\surface\Desktop\che-component-with-git\coffee-editor-master-0314\client\node_modul
es\electron-rebuild\lib\src\cli.js:158:9
Reverted "native-keymap"
Reverted "drivelist"
Reverted "find-git-repositories"
Reverted "nsfw"
Reverted "node-pty"
error Command failed with exit code 4294967295`

so l confused why `yarn rebuild:browser` success, but `yarn rebuild:electron` fail with 'Could not find any Visual Studio installation to use', thanks!

**Comment by ndoschek** [comment, marked as answer]
Hi @davidxwwang,

As the coffee-editor is based on Eclipse [Theia](https://github.com/eclipse-theia/theia), could you please check if your system fulfills the prerequisites: https://github.com/eclipse-theia/theia/blob/master/doc/Developing.md#prerequisites

HTH and best regards,
Nina

**Reply by davidxwwang** [reply]
hi @ndoschek ,thanks for you reply, l check the souceCode,and then add GYP_MSVS_VERSION as env variable, the problem fixed,

**Reply by ndoschek** [reply]
Great, I'm glad your build problem could be resolved!
